docs/source/getting_started.rst

```diff
@@ -4,84 +4,132 @@ Getting started
-should add some first, as mypy won't report many errors in unannotated
-functions. Don't worry if you aren't familiar with type annotations --
-we'll discuss them in detail in much of the rest of this guide.
+      $ mypy --py2 program.py
 
-Mypy supports many command line options that you can use to tweak how
-mypy behaves.  They are documented in :ref:`command-line`.
 
-Function signatures
-*******************
+.. note::
+
+  Depending on how mypy is configured, you may have to run mypy like this:
 
-A function without a type annotation is considered *dynamically typed* by
-mypy:
+  .. code-block:: shell
+
+    $ python3 -m mypy program.py
+
+    # Or on Windows
+    $ py -3 -m mypy program.py
+
+Function signatures and dynamic vs static typing
+************************************************
+
+A function without type annotations is considered to be *dynamically typed* by mypy:
 
 .. code-block:: python
 
    def greeting(name):
-       return 'Hello, {}'.format(name)
+       return 'Hello ' + name
+
+By default, mypy will **not** type check dynamically typed functions. This means
+that with a few exceptions, mypy will not report any errors with regular unannotated Python.
 
-You can declare the signature of a function using the Python 3
-annotation syntax (Python 2 is discussed later in :ref:`python2`).
-This makes the function statically typed, which causes mypy to
-report type errors within the function.
+This is the case even if we misuse the function: for example, mypy would currently
+not report any errors if we tried running ``greeting(3)`` or ``greeting(b"Alice")``
+even though those function calls would result in errors at runtime.
 
-Here's a version of the above function that is statically typed and
-will be type checked:
+We can teach mypy to detect these kinds of bugs by adding *type annotations* (also
+known as *type hints*). For example, we can teach mypy that ``greeting`` both accepts
+and returns a string like so:
 
 .. code-block:: python
 
    def greeting(name: str) -> str:
-       return 'Hello, {}'.format(name)
+       return 'Hello ' + name
 
-Now mypy will reject the following call, since the argument has an
-incompatible type:
+Our function is now *statically typed*: mypy can use the provided type hints to detect
+incorrect usages of the ``greeting`` function. For example, it will reject the following
+calls since the arguments have invalid types:
 
 .. code-block:: python
 
    def greeting(name: str) -> str:
-       return 'Hello, {}'.format(name)
+       return 'Hello ' + name
 
+   greeting(3)         # Argument 1 to "greeting" has incompatible type "int"; expected "str"
    greeting(b'Alice')  # Argument 1 to "greeting" has incompatible type "bytes"; expected "str"
 
+Note that this is all still valid Python 3 code! The function annotation syntax
+shown above was added to Python `as a part of Python 3.0 <pep3107_>`_.
+
+If you are trying to type-check Python 2 code, you can add type hints
```

Comment:
Style nit: Above you wrote "type check" without a dash. I prefer to write it as "type check".